File: backend/app/models/business.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import DateTime, String, Text, Uuid
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base


class Business(Base):
    __tablename__ = "businesses"

    id: Mapped[uuid.UUID] = mapped_column(Uuid(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name: Mapped[str] = mapped_column(String(255))
    # Optional white-label strings for mobile/web UI (per workspace). OS app name still from store listing.
    branding_title: Mapped[str | None] = mapped_column(String(128), nullable=True)
    branding_logo_url: Mapped[str | None] = mapped_column(String(512), nullable=True)
    gst_number: Mapped[str | None] = mapped_column(String(20), nullable=True)
    address: Mapped[str | None] = mapped_column(Text, nullable=True)
    phone: Mapped[str | None] = mapped_column(String(32), nullable=True)
    contact_email: Mapped[str | None] = mapped_column(String(255), nullable=True)
    default_currency: Mapped[str] = mapped_column(String(3), default="INR")
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

    memberships = relationship("Membership", back_populates="business")

File: backend-dotnet/src/Domain/Entities/Core/Business.cs
using System.ComponentModel.DataAnnotations.Schema;
using PurchaseAssistant.Domain.Common;

namespace PurchaseAssistant.Domain.Entities.Core;

[Table("businesses")]
public class Business : BaseEntity
{
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("branding_title")]
    public string? BrandingTitle { get; set; }

    [Column("branding_logo_url")]
    public string? BrandingLogoUrl { get; set; }

    [Column("gst_number")]
    public string? GstNumber { get; set; }

    [Column("address")]
    public string? Address { get; set; }

    [Column("phone")]
    public string? Phone { get; set; }

    [Column("contact_email")]
    public string? ContactEmail { get; set; }

    [Column("default_currency")]
    public string DefaultCurrency { get; set; } = "INR";
}
